pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -157,16 +163,19 @@ func (c *InstallerController) createInstallerController(operatorSpec *operatorv1
 // newNodeStateForInstallInProgress returns the new NodeState or error
 func (c *InstallerController) newNodeStateForInstallInProgress(currNodeState *operatorv1alpha1.NodeStatus) (*operatorv1alpha1.NodeStatus, error) {
 	ret := currNodeState.DeepCopy()
-	installerPod, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Get(getInstallerPodName(currNodeState.TargetDeploymentGeneration, currNodeState.NodeName), metav1.GetOptions{})
+	installerPod, err := c.podLister.Pods(c.targetNamespace).Get(getInstallerPodName(currNodeState.TargetDeploymentGeneration, currNodeState.NodeName))
 	if apierrors.IsNotFound(err) {
-		ret.LastFailedDeploymentGeneration = currNodeState.TargetDeploymentGeneration
-		ret.TargetDeploymentGeneration = currNodeState.CurrentDeploymentGeneration
-		ret.LastFailedDeploymentErrors = []string{err.Error()}
+		if len(ret.LastFailedDeploymentErrors) == 0 {
```

Comment:
this is somewhat controversial... this means that if we failed because of number of restart was exceeded, we deleted the pod and already set the last failed deployment errors. However, the pod won't exists anymore.